### PR TITLE
Require active channel be CLOSED before re-opening.

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

closes: #5580 

### Commit Message / Changelog Entry

```text
chore: require active channel be CLOSED before re-opening.
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by damiannolan — APPROVED** [review]
🚀

File: modules/apps/27-interchain-accounts/controller/keeper/handshake.go [hunk 1]
```diff
@@ -67,8 +67,8 @@ func (k Keeper) OnChanOpenInit(
 			panic(fmt.Errorf("active channel mapping set for %s but channel does not exist in channel store", activeChannelID))
 		}
 
-		if channel.State == channeltypes.OPEN {
-			return "", errorsmod.Wrapf(icatypes.ErrActiveChannelAlreadySet, "existing active channel %s for portID %s is already OPEN", activeChannelID, portID)
+		if channel.State != channeltypes.CLOSED {
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/controller/keeper/handshake.go:70]: This is needed on the host side as well?

Comment (damiannolan) [reply modules/apps/27-interchain-accounts/controller/keeper/handshake.go:70]: yeah I think the host side still uses the `channel.IsOpen()` func

Comment (crodriguezvega) [reply modules/apps/27-interchain-accounts/controller/keeper/handshake.go:70]: Yes, [it does](https://github.com/cosmos/ibc-go/blob/a7f85d5e037da02428f23c493b4584b088ef2c02/modules/apps/27-interchain-accounts/host/keeper/handshake.go#L57-L59).

Comment (crodriguezvega) [reply modules/apps/27-interchain-accounts/controller/keeper/handshake.go:70]: Opened https://github.com/cosmos/ibc-go/issues/5629.

**Pull request merged**